sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLCLIDriver.scala

```diff
@@ -278,8 +278,16 @@ private[hive] object SparkSQLCLIDriver extends Logging {
 
     var ret = 0
     var prefix = ""
-    val currentDB = ReflectionUtils.invokeStatic(classOf[CliDriver], "getFormattedDb",
-      classOf[HiveConf] -> conf, classOf[CliSessionState] -> sessionState)
+
+    def currentDB: String = {
+      val printCurrentDb = conf.getBoolVar(HiveConf.ConfVars.CLIPRINTCURRENTDB) ||
+        SparkSQLEnv.sqlContext.conf.cliPrintCurrentDb
+      if (printCurrentDb) {
+        s" (${SparkSQLEnv.sqlContext.sparkSession.catalog.currentDatabase})"
+      } else {
+        ""
```

Comment:
I'm not sure when users don't want to see the current db. If the hive config is true by default, it seems OK to not add a new conf in Spark but just respect that hive conf.

Reply:
Hive disables showing the current database for `hive-cli`, but enables it for hive `beeline` by default. As the former is a deprecated use case of hive, I would like to respect the latter for our SQL shell.

Reply:
to be safe, shall we use the old code if the config is false?

Reply:
OK

Reply:
for old code, I mean `ReflectionUtils.invokeStatic(classOf[CliDriver], "getFormattedDb", ...`, sorry for the confusion.

Reply:
do you mean 
```scala
      if (SparkSQLEnv.sqlContext.conf.cliPrintCurrentDb) {
        s" (${SparkSQLEnv.sqlContext.sparkSession.catalog.currentDatabase})"
      } else {
        ReflectionUtils.invokeStatic(classOf[CliDriver], "getFormattedDb",...
      }
```

Reply:
yup

Reply:
then, if spark disables and hive enables, then the current db will always be `default` which is a wrong answer mostly